app/src/main/java/com/mealsmadeeasy/ui/home/profile/ProfileFragment.kt

```diff
@@ -0,0 +1,203 @@
+package com.mealsmadeeasy.ui.home.profile
+
+import android.app.AlertDialog
+import android.app.DatePickerDialog
+import android.content.Context
+import android.os.Bundle
+import android.support.design.widget.TextInputLayout
+import android.text.Editable
+import android.text.TextWatcher
+import android.text.format.DateFormat
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.*
+import com.mealsmadeeasy.R
+import com.mealsmadeeasy.ui.BaseFragment
+
+import java.util.*
+import android.widget.ArrayAdapter
+import com.mealsmadeeasy.model.Gender
+import com.mealsmadeeasy.model.UserProfile
+
+
+private const val MIN_AGE = 13
+private const val INCHES_IN_FEET = 12
+private const val MIN_BMI = 15
+private const val MAX_BMI = 50
+
+class ProfileFragment : BaseFragment() {
+    lateinit var sexSpinner: Spinner
+    lateinit var bdayPicker: EditText
+    lateinit var feetText: EditText
+    lateinit var inchesText: EditText
+    lateinit var weightText: EditText
+    lateinit var submitButton: Button
+
+    companion object {
+        fun newInstance(): ProfileFragment = ProfileFragment()
+    }
+
+    fun heightToInches(feet: Int, inches: Int): Int {
+        return feet * INCHES_IN_FEET + inches
+    }
+
+    private fun areAllFieldsEnabled(): Boolean {
```

Comment:
Can we move this below some of the override methods? I like to see the juicy stuff first when I open a class.